sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -195,28 +216,71 @@ object RmmRapidsRetryIterator extends Arm {
   private case class SingleItemAutoCloseableIteratorInternal[T <: AutoCloseable](ts: T)
     extends Iterator[T] with AutoCloseable {
 
-    private var wasCalled = false
-    override def hasNext: Boolean = !wasCalled
+    private var wasCalledSuccessfully = false
+    override def hasNext: Boolean = !wasCalledSuccessfully
     override def next(): T = {
-      wasCalled = true
+      wasCalledSuccessfully = true
       ts
     }
     override def close(): Unit = {
-      if (!wasCalled) {
+      if (!wasCalledSuccessfully) {
         ts.close()
       }
     }
   }
 
   /**
-   * RmmRapidsRetryAutoCloseableIterator exposes an iterator that can retry work,
-   * specified by `fn`, abstracting away the retry specifics. Elements passed to this iterator
-   * must be AutoCloseable.
+   * A trait that defines an iterator of type K that supports two extra things:
+   * the ability to split its input, and the ability to close itself.
```

Comment:
nit: can we have some more docs defining what split means and what it is supposed to do?  I understand it because of how it is used, but I realized the interface itself does not describe that.

Reply:
https://github.com/NVIDIA/spark-rapids/pull/7852/commits/dd1bad306cb9f7e371a0601b65cb7f9b062be42f added a scaladoc header for it.